src/main/java/mat/server/util/MeasureUtility.java

```diff
@@ -40,8 +40,19 @@ public static String getVersionText(String version, boolean isDraft) {
     public static String formatVersionText(String version) {
         StringUtility su = new StringUtility();
         String[] versionArr = version.split("\\.");
-        String majorVersion = su.trimLeadingZeros(versionArr[0]);
-        String minorVersion = su.trimLeadingZeros(versionArr[1]);
+        String majorVersion = "";
+        String minorVersion = "";
+        String revisionNumber = "";
+        for(int i = 0; i < versionArr.length; i++) {
+            if(i == 0) {
+                majorVersion = su.trimLeadingZeros(versionArr[0]);
+            } else if(i == 1) {
+                minorVersion = su.trimLeadingZeros(versionArr[1]);
+            } else {
+                revisionNumber = su.trimLeadingZeros(versionArr[2]);
+                return majorVersion + "." + minorVersion + "." + revisionFormat.format(Integer.parseInt(revisionNumber));
+            }

```

Comment:
I think this block could be condensed; looks like there are only 2 cases:
```
String majorVersion = su.trimLeadingZeros(versionArr[0]);
String minorVersion = su.trimLeadingZeros(versionArr[1]);
if (versionArr.length > 2) 
    return majorVersion + "." + minorVersion + "." + revisionFormat.format(Integer.parseInt(su.trimLeadingZeros(versionArr[2]));

return majorVersion + "." + su.trimLeadingZeros(versionArr[1]);
```

For revisionFormat, is there a benefit to converting the String to an int then formatting back to a String?

Reply:
It is to make sure that at versionArr[0] / [1] it doesn't give out of bound exception.

Reply:
Ah, are we assuming the incoming version is not well formed?

Does the calling method know how to handle a single dot, "3.", or "1..3"?

Reply:
Nope, it will display it as "1..3".
Ideally, the exiting code is exactly the same as you mentioned, which is 
```
String majorVersion = su.trimLeadingZeros(versionArr[0]);
String minorVersion = su.trimLeadingZeros(versionArr[1]);
if (versionArr.length > 2) 
    return majorVersion + "." + minorVersion + "." + revisionFormat.format(Integer.parseInt(su.trimLeadingZeros(versionArr[2]));

return majorVersion + "." + su.trimLeadingZeros(versionArr[1]);
```
I updated based on last CR comments to handle exception.